src/streams/mbedtls.c

```diff
@@ -70,6 +69,7 @@ static void shutdown_ssl(void)
 		git__free(mbedtls_entropy);
 		mbedtls_entropy = NULL;
 	}
+	git__free(git__cipher_list);
```

Comment:
You should re-initialize the value, just to make sure we never run into a double free